xdocs/usermanual/properties_reference.xml

```diff
@@ -1339,6 +1339,14 @@ JMETER-SERVER</source>
     ORO PatternCacheLRU size.<br/>
     Defaults to: <code>1000</code>
 </property>
+<property name="function.cache.per.iteration">
+    <p>Cache function execution during test execution.</p>
+    <p>By default, JMeter caches function properties during a test iteration, however,
+    it might cause unexpected results when a component is shared across threads and the expression depends on
+    the thread variables.</p>
+    <note>The caching behaviour would likely be disabled in the upcoming versions</note>
```

Comment:
```suggestion
    <note>The caching behaviour will likely be disabled in an upcoming version</note>
```
Looks good to me.
Out of curiosity, have you done performance comparisons? I would think, that in most cases on modern CPUs the cache is not that important anymore.

Reply:
I believe the cache creates functional issues: it caches the values, and it does not invalidate the cache. In other words, it assumes "the computed value depends only on the iteration number", which, in my opinion, is quite poor approximation.

So my main goal here is to establish correctness, and if there are any performance implications, then they should be healed separately

Reply:
I have not done explicit comparisons, however, I saw no perf difference when it comes to my regular load testing results